Fix Event Next/Last Match

Shouldn't have this case...
![screenshot_20160227-180856](https://cloud.githubusercontent.com/assets/2754863/13380398/16647e40-de10-11e5-87bd-d422626d3c80.png)

May relate to the following stack traces:
```
java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.Worker thread.
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:62)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5466)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.view.View.setVisibility(int)' on a null object reference
	at com.thebluealliance.androidclient.binders.EventInfoBinder.lambda$onEvent$35(EventInfoBinder.java:248)
	at com.thebluealliance.androidclient.binders.EventInfoBinder.access$lambda$1(EventInfoBinder.java)
	at com.thebluealliance.androidclient.binders.EventInfoBinder$$Lambda$2.call(Unknown Source)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	... 7 more
```

```
java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.Worker thread.
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:62)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5343)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:905)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:700)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.FrameLayout.setVisibility(int)' on a null object reference
	at com.thebluealliance.androidclient.binders.EventInfoBinder.showLastMatch(EventInfoBinder.java:216)
	at com.thebluealliance.androidclient.binders.EventInfoBinder.lambda$onEvent$34(EventInfoBinder.java:233)
	at com.thebluealliance.androidclient.binders.EventInfoBinder.access$lambda$0(EventInfoBinder.java)
	at com.thebluealliance.androidclient.binders.EventInfoBinder$$Lambda$1.call(Unknown Source)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	... 8 more
```

--- a/android/src/main/java/com/thebluealliance/androidclient/binders/EventInfoBinder.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/binders/EventInfoBinder.java
@@ -36,6 +36,7 @@ public class EventInfoBinder extends AbstractDataBinder<EventInfoBinder.Model> {
     private LayoutInflater mInflater;
     private MatchRenderer mMatchRenderer;
     private boolean mIsLive;
+    private boolean mAreViewsBound;
 
     @Inject SocialClickListener mSocialClickListener;
     @Inject EventInfoContainerClickListener mInfoClickListener;
@@ -70,6 +71,8 @@ public class EventInfoBinder extends AbstractDataBinder<EventInfoBinder.Model> {
         mSocialClickListener = socialClickListener;
         mInfoClickListener = eventInfoContainerClickListener;
         mMatchRenderer = renderer;
+        mIsLive = false;
+        mAreViewsBound = false;
     }
 
     public void setInflater(LayoutInflater inflater) {
@@ -78,7 +81,10 @@ public class EventInfoBinder extends AbstractDataBinder<EventInfoBinder.Model> {
 
     @Override
     public void bindViews() {
-        ButterKnife.bind(this, mRootView);
+        if (!mAreViewsBound) {
+            ButterKnife.bind(this, mRootView);
+            mAreViewsBound = true;
+        }
     }
 
     @Override
@@ -189,6 +195,7 @@ public class EventInfoBinder extends AbstractDataBinder<EventInfoBinder.Model> {
         super.unbind(unbindViews);
         if (unbindViews) {
             ButterKnife.unbind(this);
+            mAreViewsBound = false;
         }
     }
 
@@ -215,6 +222,9 @@ public class EventInfoBinder extends AbstractDataBinder<EventInfoBinder.Model> {
     }
 
     protected void showLastMatch(MatchListElement match) {
+        if (!mAreViewsBound) {
+            bindViews();
+        }
         lastMatchView.setVisibility(View.VISIBLE);
         lastMatchContainer.setVisibility(View.VISIBLE);
         lastMatchView.removeAllViews();
@@ -222,22 +232,45 @@ public class EventInfoBinder extends AbstractDataBinder<EventInfoBinder.Model> {
     }
 
     protected void showNextMatch(MatchListElement match) {
+        if (!mAreViewsBound) {
+            bindViews();
+        }
         nextMatchView.setVisibility(View.VISIBLE);
         nextMatchContainer.setVisibility(View.VISIBLE);
         nextMatchView.removeAllViews();
         nextMatchView.addView(match.getView(mActivity, mInflater, null));
     }
 
+    protected void hideLastMatch() {
+        if (!mAreViewsBound) {
+            bindViews();
+        }
+        lastMatchContainer.setVisibility(View.GONE);
+    }
+
+    protected void hideNextMatch() {
+        if (!mAreViewsBound) {
+            bindViews();
+        }
+        nextMatchContainer.setVisibility(View.GONE);
+    }
+
     @SuppressWarnings("unused")
     public void onEvent(LiveEventMatchUpdateEvent event) {
         AndroidSchedulers.mainThread().createWorker().schedule(() -> {
-            if (mIsLive && event.getLastMatch() != null) {
+            if (mIsLive && event != null && event.getLastMatch() != null) {
                 Log.d(Constants.LOG_TAG, "showing last match");
                 showLastMatch(mMatchRenderer.renderFromModel(event.getLastMatch(), RENDER_DEFAULT));
+            } else {
+                Log.d(Constants.LOG_TAG, "hiding last match");
+                hideLastMatch();
             }
-            if (mIsLive && event.getNextMatch() != null) {
+            if (mIsLive && event != null && event.getNextMatch() != null) {
                 Log.d(Constants.LOG_TAG, "showing next match");
                 showNextMatch(mMatchRenderer.renderFromModel(event.getNextMatch(), RENDER_DEFAULT));
+            } else {
+                Log.d(Constants.LOG_TAG, "hiding next match");
+                hideNextMatch();
             }
         });
     }

